soroban-env-common/src/array.rs

```diff
@@ -1,25 +1,28 @@
+use crate::xdr::ScHostValErrorCode;
 use stellar_xdr::ScObjectType;
 
 use crate::{
-    ConversionError, Env, EnvVal, IntoVal, Object, RawVal, RawValConvertible, TryFromVal,
-    TryIntoVal,
+    ConversionError, Env, Object, RawVal, RawValConvertible, Status, TryFromVal, TryIntoVal,
 };
 
+// TODO: these conversions happen as RawVal, but they actually take and produce
+// Objects; consider making the signatures tighter.
+
 impl<E: Env, const N: usize> TryFromVal<E, RawVal> for [u8; N] {
-    type Error = ConversionError;
+    type Error = Status;
```

Comment:
This also means we're introducing non zero sized errors into the SDK for type conversions. I'm fine with it, but I thought a few people including yourself had an issue with doing that. Maybe we should use Status instead of ConversionError, everywhere?